pkg/build/buildkit/build.go

```diff
@@ -252,6 +268,29 @@ func (b *BuildKit) extractTsuruConfigsFromContainerImage(ctx context.Context, c
 	return callBuildKitToExtractTsuruConfigs(ctx, c, tmpDir, workingDir)
 }
 
+func (b *BuildKit) createRemoteRepository(ctx context.Context, r *pb.BuildRequest) error {
+	for _, v := range r.DestinationImages {
+		if provider, ok := b.opts.RemoteRepository[build.GetRegistry(v)]; ok {
+			err := provider.Auth(ctx)
+			if err != nil {
+				return err
```

Comment:
@morpheu I suggest to you simplify Provider interface in just one method: Ensure()